llama_index/vector_stores/elasticsearch.py

```diff
@@ -21,6 +24,20 @@
     "EUCLIDEAN_DISTANCE",
 ]
 
+nest_asyncio.apply()
+
+
+def add_sync_version(func: Any) -> Any:
+    """Decorator for adding sync version of an async function."""
+    assert asyncio.iscoroutinefunction(func)
+
+    @wraps(func)
+    def _wrapper(self: Any, *args: Any, **kwds: Any) -> Any:
+        return asyncio.get_event_loop().run_until_complete(func(self, *args, **kwds))
+
+    func.sync = _wrapper
+    return func
+
```

Comment:
very nit: let's move this into an utils file? Would be useful as we adopt this approach in other parts of the codebase.

Reply:
So, I made (hopefully the last modification on this PR...):

- instead of using a decorator, i directly use `asyncio.get_event_loop().run_until_complete(self.ASYNC_FUNC())` within the `sync` methods `add`, `delete` and `query`.
- the decorator seems a bit redundant here and so i think the current change is simple and easier to understand.
- i did however also move this function into `utils.py` anyhow just in case we might want to use it to decorate standalone async functions (to get sync versions) elsewhere in this library